### PR TITLE
Laying groundwork for fuzz-testing and sanitizer profiles

### PR DESCRIPTION
**Pull request by beckerhe**
This PR adds conan-settings for enabling sanitizers including libfuzzer.

Furthermore it adjust the LLVM packages which needed some changes to be able to be compiled with sanitizers enabled.

Additionally this PR makes some changes on the way:
- Removes debug output from LLVM conan packages. This should make a build log less noisy.
- Replace the zlib-package with the new official one.
- Adds the latest GCC version numbers to our settings.yml. (Not needed currently.)
- Backports a bugfix from LLVM master which fixes a compatibility issue with Visual Studio 2019 15.6

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -58,13 +58,13 @@ def requirements(self):
         self.requires(
             "grpc/1.27.3@{}#dc2368a2df63276188566e36a6b7868a".format(self._orbit_channel))
         self.requires("gtest/1.8.1@bincrafters/stable#0")
-        self.requires("llvm_object/9.0.1@orbitdeps/stable#0")
+        self.requires("llvm_object/9.0.1-1@orbitdeps/stable")
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply conanfile.py:61]: We could. And I'm still debating with myself it it's a good idea in this case. Maybe we should. I'll have a look.

File: third_party/conan/recipes/llvm-common/llvmcomponentpackage.py [hunk 2]
```diff
@@ -89,6 +96,9 @@ def build(self):
         cmake_options['LLVM_ENABLE_TERMINFO'] = False
         cmake_options['LLVM_ENABLE_RTTI'] = not self.options.no_rtti
 
+        if self.options.allow_undefined_symbols:
+            cmake_options['ALLOW_UNDEFINED_SYMBOLS'] = True
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply third_party/conan/recipes/llvm-common/llvmcomponentpackage.py:100]: Do you happen to know whether that works?

File: conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (beckerhe) [reply conanfile.py]: Added the revision. Thanks.

File: third_party/conan/recipes/llvm-common/llvmcomponentpackage.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (beckerhe) [reply third_party/conan/recipes/llvm-common/llvmcomponentpackage.py]: Seems to work. Done.

**Comment by beckerhe** [comment]
Packages are currently being build. Should be ready by tomorrow.

**Comment by beckerhe** [comment]
I finally managed to fix all the lockfiles.

The following changes have been done in addition to what has been mentioned before:

- The `gen_lockfiles.sh` script creates now a "full" lockfile, containing all transitive dependencies from 3rd-party packages. This was not the case before.
- The lockfiles have been updated to contain the full build-graph, including all the build requirements for all the dependencies (this is due to the change in `gen_lockfiles.sh`)

**Comment by beckerhe** [comment]
@IrinaShkviro PTAL. Thanks. :-)

File: third_party/conan/lockfiles/linux/clang7_debug/conan.lock [hunk 3]
```diff
@@ -306,44 +425,868 @@
...
+    "options": ""
+   },
+   "43": {
+    "pref": "cmake_installer/3.16.3@conan/stable#0:44fcf6b9a7fb86b2586303e3db40189d3b511830",
+    "options": ""
+   },
+   "44": {
+    "pref": "cmake_installer/3.16.3@conan/stable#0:44fcf6b9a7fb86b2586303e3db40189d3b511830",
+    "options": ""
+   },
```
Comment (IrinaShkviro) [inline third_party/conan/lockfiles/linux/clang7_debug/conan.lock:485]: The nodes from 39 to 44 are the same. Is it expected?

File: third_party/conan/recipes/llvm/test_package/conanfile.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class DefaultNameConan(ConanFile):
     name = "DefaultName"
     version = "0.1"
     settings = "os", "compiler", "arch", "build_type"
-    requires = "llvm/9.0.1@%s/%s" % (username, channel)
+    requires = "llvm/9.0.1-1@%s/%s" % (username, channel)
```
Comment (IrinaShkviro) [inline third_party/conan/recipes/llvm/test_package/conanfile.py:18]: Shouldn't be `llvm/9.0.1-2`?

File: third_party/conan/lockfiles/linux/clang7_debug/conan.lock (hunk 3 above)
Comment (beckerhe) [reply third_party/conan/lockfiles/linux/clang7_debug/conan.lock:485]: Yes. Every package get it's own node with build requirements.

File: third_party/conan/recipes/llvm/test_package/conanfile.py (hunk 4 above)
Comment (beckerhe) [reply third_party/conan/recipes/llvm/test_package/conanfile.py:18]: Yes thank. Will fix that!

Comment (beckerhe) [reply third_party/conan/recipes/llvm/test_package/conanfile.py]: done.

**Pull request merged**